source/agora/common/Deserializer.d

```diff
@@ -110,11 +109,32 @@ unittest
 {
     import agora.common.Hash;
     import agora.common.Serializer;
+    import agora.consensus.Genesis;
 
-    scope Block block = getGenesisBlock();
+    const block = getGenesisBlock();
 
     // Block Testing serialization & deserialization
     // Compare the serialization data with the origin Ledger data.
     ubyte[] serializedData = serializeFull(block);
     assert(block == deserializeBlock(serializedData));
+
+    // Check that there is no trailing data
+    ubyte[] arrayData = serializeFull(block) ~ serializeFull(block);
+
+    void deserializeArrayEntry () nothrow @safe
+    {
+        scope DeserializeDg dg = (size) nothrow @safe
+        {
+            ubyte[] res = arrayData[0 .. size];
+            arrayData = arrayData[size .. $];
+            return res;
+        };
+
+        Block nblock;
+        nblock.deserialize(dg);
+        assert(nblock == nblock);
```

Comment:
Isn't this always true?

Reply:
Oups that should be `block == nblock`, good eye